src/components/accounts/SetupImplicit.js

```diff
@@ -0,0 +1,289 @@
+            })
+        }
+    }
+
+    componentDidMount = () => {
+        this.connection = nearApiJs.Connection.fromConfig({
+            networkId: NETWORK_ID,
+            provider: { type: 'JsonRpcProvider', args: { url: NODE_URL + '/' } },
+            signer: {},
+        })
+
+        const recoveryKeyPair = JSON.parse(localStorage.getItem(IMPLICIT_ACCOUNT_KEY) || '{}')
+
+        if (!recoveryKeyPair.publicKey) {
+            if (this.props.location.pathname === '/create-implicit') {
+                this.refreshData()
+            }
+        } else {
+            this.setState({ recoveryKeyPair })
+        }
+    }
+
+    refreshData = () => {
+        const { seedPhrase, publicKey, secretKey } = generateSeedPhrase()
+        const recoveryKeyPair = KeyPair.fromString(secretKey)
+        const wordId = Math.floor(Math.random() * 12)
+
+        this.setState((prevState) => ({
+            ...prevState,
+            seedPhrase,
+            publicKey,
+            wordId,
+            enterWord: '',
+            requestStatus: null,
+            recoveryKeyPair
+        }))
+    }
+
+    handleChangeWord = (e, { name, value }) => {
+        if (value.match(/[^a-zA-Z]/)) {
+            return false
+        }
+
+        this.setState((state) => ({
+           [name]: value.trim().toLowerCase(),
+           requestStatus: null
+        }))
+    }
+
+    handleStartOver = e => {
+        const {
+            history
+        } = this.props
+
+        this.refreshData()
+        history.push(`/create-implicit`)
+    }
+
+    handleSubmit = async () => {
+        const {
+            history
+        } = this.props
+        const { seedPhrase, enterWord, wordId, recoveryKeyPair } = this.state
+        if (enterWord !== seedPhrase.split(' ')[wordId]) {
+            this.setState(() => ({
+                requestStatus: {
+                    success: false,
+                    messageCode: 'account.verifySeedPhrase.error'
+                }
+            }))
+            return false
+        }
+        console.log('verified', recoveryKeyPair)
+        recoveryKeyPair.seedPhrase = seedPhrase
+        recoveryKeyPair.implicitAccountId = Buffer.from(recoveryKeyPair.publicKey.data).toString('hex')
+        localStorage.setItem(IMPLICIT_ACCOUNT_KEY, JSON.stringify(recoveryKeyPair))
+        history.push(`/create-implicit/fund`)
+    }
+
+    handleCopyPhrase = (textToCopy) => {
+        if (typeof textToCopy !== 'string') {
+            textToCopy = null
+        }
+        if (navigator.share && isMobile()) {
+            navigator.share({
+                text: textToCopy ? textToCopy : this.state.seedPhrase
+            }).catch(err => {
+                console.log(err.message);
+            });
+        } else {
+            this.handleCopyDesktop(textToCopy);
+        }
+    }
+
+    handleCopyDesktop = (textToCopy) => {
+        copyText(textToCopy ? document.getElementById('implicit-account-id') : document.getElementById('seed-phrase'));
```

Comment:
not sure I understand the logic with `textToCopy`. May you explain?

Reply:
the functions were purpose built to copy seed phrase, I'm just trying to generalize so I can copy seed phrase as well.

Reply:
added comment